flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/metrics/MySqlSourceReaderMetrics.java

```diff
@@ -44,6 +69,49 @@ public void registerMetrics() {
                 MetricNames.CURRENT_FETCH_EVENT_TIME_LAG, (Gauge<Long>) this::getFetchDelay);
     }
 
+    public void numRecordsOutSnapshotIncrease(TableId tableId) {
+        Counter counter =
+                snapshotNumRecordsOutMap.computeIfAbsent(
+                        tableId,
+                        k ->
+                                metricGroup.counter(
+                                        tableId.identifier() + IO_NUM_RECORDS_OUT_SNAPSHOT));
+        counter.inc();
+    }
+
+    public void numRecordsOutInsertIncrease(TableId tableId) {
+        Counter counter =
+                insertNumRecordsOutMap.computeIfAbsent(
+                        tableId,
+                        k ->
+                                metricGroup.counter(
+                                        tableId.identifier()
+                                                + IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_INSERT));
+        counter.inc();
+    }
+
+    public void numRecordsOutUpdateIncrease(TableId tableId) {
+        Counter counter =
+                updateNumRecordsOutMap.computeIfAbsent(
+                        tableId,
+                        k ->
+                                metricGroup.counter(
+                                        tableId.identifier()
+                                                + IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_UPDATE));
+        counter.inc();
+    }
+
+    public void numRecordsOutDeleteIncrease(TableId tableId) {
+        Counter counter =
+                deleteNumRecordsOutMap.computeIfAbsent(
+                        tableId,
+                        k ->
+                                metricGroup.counter(
+                                        tableId.identifier()
+                                                + IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_DELETE));
+        counter.inc();
+    }
+
```

Comment:
I applied this metrics my code, and left some comments.

- Metric names should be general. 
    - In your case, It generate uniqe metric for each tables. But in monitorig tool such as grafana, we distingush tables by property in metric, not by mtric name
So I suggest to remove tableId.   
```suggestion
    public void numRecordsOutSnapshotIncrease(TableId tableId) {
        Counter counter =
                snapshotNumRecordsOutMap.computeIfAbsent(
                        tableId,
                        k ->
                                metricGroup.counter(IO_NUM_RECORDS_OUT_SNAPSHOT));
        counter.inc();
    }

    public void numRecordsOutInsertIncrease(TableId tableId) {
        Counter counter =
                insertNumRecordsOutMap.computeIfAbsent(
                        tableId,
                        k ->
                                metricGroup.counter(IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_INSERT));
        counter.inc();
    }

    public void numRecordsOutUpdateIncrease(TableId tableId) {
        Counter counter =
                updateNumRecordsOutMap.computeIfAbsent(
                        tableId,
                        k ->
                                metricGroup.counter(IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_UPDATE));
        counter.inc();
    }

    public void numRecordsOutDeleteIncrease(TableId tableId) {
        Counter counter =
                deleteNumRecordsOutMap.computeIfAbsent(
                        tableId,
                        k ->
                                metricGroup.counter(IO_NUM_RECORDS_OUT_DATA_CHANGE_EVENT_DELETE));
        counter.inc();
    }
```

Reply:
Ok, thanks for reviewing the code for me, but the original purpose of this metric is to see each table at a granular level, if you need to see the total, I will add other metrics